src/WebJobs.Script/Binding/FunctionBinding.cs

```diff
@@ -213,7 +213,7 @@ internal static ICollection ReadAsCollection(object value)
             else
             {
                 // Read the value into the context Value converting based on data type
-                object converted = null;
+                object converted = context.Value;
```

Comment:
Could `Value` be null? If so, you may want to have something like `object converted = context.Value ?? new MemoryStream()`